src/client/index.ts

```diff
@@ -573,8 +573,18 @@ export class Client {
 		if (this.ui) {
 			this.ui.showLoaderDelayed([
 				"<h4>Authenticating...</h4>",
-				"<small>You may need to sign a new challenge in your wallet</small>"
+				"<small>You may need to sign a new challenge in your wallet</small>",
+				"<button class='cancel-auth-btn' aria-label='Cancel authentication'>Cancel</button>"
 			], 600, true);
+
+			waitForElementToExist('.cancel-auth-btn').then((cancelAuthButton) => {
+				cancelAuthButton.addEventListener('click', () => {
```

Comment:
To simplify this and use JS internal event listening should we use

```` 
cancelAuthButton.onClick = ...
````

Reply:
@nicktaras sure let me update that